target/linux/mediatek/patches-6.6/341-mtd-spinand-add-support-for-HeYangTek-HYF1GQ4UDACAE.patch

```diff
@@ -0,0 +1,155 @@
+From bcda7d75e60c1a433f7146e9d30951798c1e78c3 Mon Sep 17 00:00:00 2001
+From: Maxim Anisimov <maxim.anisimov.ua@gmail.com>
+Date: Fri, 24 May 2024 11:03:17 +0300
+Subject: [PATCH] mtd: spinand: add support for HeYangTek HYF1GQ4UDACAE
+
+Add support for HeYangTek HYF1GQ4UDACAE SPI NAND.
+Datasheet:
+  https://www.heyangtek.cn/previewfile.jsp?file=ABUIABA9GAAgwsvRnwYo-eDpsgc
+
+This commit is based on https://github.com/keenetic/kernel-49/commit/bacade569fb12bc0ad31ba09bca9b890118fbca7
+
+Signed-off-by: Maxim Anisimov <maxim.anisimov.ua@gmail.com>
+---
+ drivers/mtd/nand/spi/Makefile    |   2 +-
+ drivers/mtd/nand/spi/core.c      |   1 +
+ drivers/mtd/nand/spi/heyangtek.c | 104 +++++++++++++++++++++++++++++++
+ include/linux/mtd/spinand.h      |   1 +
```

Comment:
Please send this patch upstream too. It should get integrated into mainline Linux kernel.

Please move this file to `target/linux/generic/pending-6.6/489--mtd-spinand-add-support-for-HeYangTek-HYF1GQ4UDACAE.patch` so all targets get support for this new flash chip. It is likely that more vendors will use it in the future.